File: WorldFeed/Server/src/Services/Message/WorldFeed.Message.API/DTO/GetMessageDTO.cs
namespace WorldFeed.Message.DTO
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.Events;

    public class GetMessageDTO : IGetMessageDTO
    {
        [JsonProperty("event")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageEventDTO MessageEvent { get; set; }

        [JsonProperty("apps")]
        public Dictionary<long, IApp> Apps { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/GetMessageDTO.ts
import {IApp} from "../../Public/Models/Interfaces/IApp";
import {IGetMessageDTO} from "../../Public/Models/Interfaces/DTO/IGetMessageDTO";
import {IMessageEventDTO} from "../../Public/Models/Interfaces/DTO/Events/IMessageEventDTO";
import Dictionary from "typescript-dotnet-commonjs/System/Collections/Dictionaries/Dictionary";

export class GetMessageDTO implements IGetMessageDTO {
  // [JsonProperty("event")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public messageEvent: IMessageEventDTO;

  // [JsonProperty("apps")]
  public apps: Dictionary<number, IApp>;
}
